src/Components/Components/src/Binding/CascadingModelBinder.cs

```diff
@@ -11,7 +12,7 @@ namespace Microsoft.AspNetCore.Components;
 /// <summary>
 /// Defines the binding context for data bound from external sources.
 /// </summary>
-public sealed class CascadingModelBinder : IComponent, ICascadingValueComponent, IDisposable
+public sealed class CascadingModelBinder : IComponent, ICascadingValueSupplierFactory, ICascadingValueSupplier, IDisposable
```

Comment:
What's the main idea behind splitting `ICascadingValueSupplierFactory` from `ICascadingValueSupplier` rather than have them as a single thing?

Reply:
The original thinking was that `ICascadingValueSupplier` is only capable of providing a single value (at least, before my recent changes). Therefore, in order to support providing multiple values, an `ICascadingValueSupplierFactory` could create multiple `ICascadingValueSupplier` instances, one per each value it supplies.

This did lead to some downsides:
* Extra allocations for each `ICascadingValueSupplier` that gets created.
* If multiple suppliers "notify" the same component, that component will get its `SetParametersAsync` method called multiple times.

Therefore, I've updated my implementation to change `ICascadingValueSupplier` to not have a single `CurrentValue`, but instead a `GetCurrentValue()` method whose argument provides information needed to distinguish between multiple possible values, if necessary.